substrate/client/transaction-pool/src/graph/pool.rs

```diff
@@ -61,26 +62,20 @@ pub type ValidatedTransactionFor<A> =
 	ValidatedTransaction<ExtrinsicHash<A>, ExtrinsicFor<A>, <A as ChainApi>::Error>;
 
 /// Concrete extrinsic validation and query logic.
+#[async_trait]
 pub trait ChainApi: Send + Sync {
 	/// Block type.
 	type Block: BlockT;
 	/// Error type.
 	type Error: From<error::Error> + error::IntoPoolError;
-	/// Validate transaction future.
-	type ValidationFuture: Future<Output = Result<TransactionValidity, Self::Error>> + Send + Unpin;
```

Comment:
I wonder why these were configurable in the first place :thinking:. Maybe for testing?